### PR TITLE
refactor(bundle): kustomize changes

### PR DESCRIPTION
**Pull request by SteveMattar**
This is relevant only for deploying submariner using the bundle.

When working with `subctl` the manifests definition and objects creation are handled automatically, but when it comes to the operator bundle we have to make sure those resources are aligned.

* Add the missing broker manifests.
* Update the CSV description and samples.

**Note**: The `submariner-k8s-broker` namespace cannot be created during the operator bundle installation.
The user has to make sure the namespace exists before installing the bundle.
Same for the  SCC on Openshift.

Backported in #1398

### CODE REVIEW

File: config/samples/submariner_v1alpha1_servicediscovery.yaml [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
-  brokerK8sApiServer: "192.168.67.110:8443"
-  brokerK8sApiServerToken: "$(SECRET_API_SERVER_TOKEN)"
-  brokerK8sRemoteNamespace: "submariner-broker"
-  brokerK8sCA: "$(SECRET_CA)"
-  repository: "quay.io/submariner"
+  brokerK8sApiServer: "$(BROKER_K8S_API_SERVER)"
+  brokerK8sApiServerToken: "$(BROKER_K8S_API_TOKEN)"
+  brokerK8sRemoteNamespace: "$(SUBMARINER_BROKER_NAMESPACE)"
+  brokerK8sCA: "$(BROKER_K8S_CA)"
+  repository: repo
```
Comment (mangelajo) [inline config/samples/submariner_v1alpha1_servicediscovery.yaml:15]: Isn't it better to keep quay.io/submariner here ?

File: config/samples/submariner_v1alpha1_servicediscovery.yaml [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+  namespace: "$(SUBMARINER_OPERATOR_NAMESPACE)"
+  clusterID: "cluster1"
   debug: false
   globalnetEnabled: true
-  brokerK8sApiServer: "192.168.67.110:8443"
-  brokerK8sApiServerToken: "$(SECRET_API_SERVER_TOKEN)"
-  brokerK8sRemoteNamespace: "submariner-broker"
-  brokerK8sCA: "$(SECRET_CA)"
-  repository: "quay.io/submariner"
+  brokerK8sApiServer: "$(BROKER_K8S_API_SERVER)"
```
Comment (mangelajo) [inline config/samples/submariner_v1alpha1_servicediscovery.yaml]: Should we may be keep the original example? (as we don't accept https/http in front I think it's a little bit weird)

File: config/samples/submariner_v1alpha1_submariner.yaml [hunk 3]
```diff
@@ -2,26 +2,29 @@
...
 spec:
-  namespace: "submariner-operator"
-  serviceCIDR: "192.168.66.0/24"
-  clusterCIDR: "192.168.67.0/24"
-  globalCIDR: "169.254.0.0/16"
-  clusterID: "east"
-  colorCodes: "blue"
+  serviceCIDR: ""
+  clusterCIDR: ""
+  globalCIDR: "169.254.1.0/24"
```
Comment (mangelajo) [inline config/samples/submariner_v1alpha1_submariner.yaml]: I would also keep the global CIDR empty by default. (globalnet disabled)

File: config/samples/submariner_v1alpha1_submariner.yaml [hunk 4]
```diff
@@ -2,26 +2,29 @@
...
+  clusterID: "cluster1"
   debug: false
   natEnabled: true
   serviceDiscoveryEnabled: true
   broker: "k8s"
-  brokerK8sApiServer: "192.168.67.110:8443"
-  brokerK8sApiServerToken: "$(SECRET_API_SERVER_TOKEN)"
-  brokerK8sRemoteNamespace: "submariner-broker"
-  brokerK8sCA: "$(SECRET_CA)"
+  brokerK8sApiServer: "$(BROKER_K8S_API_SERVER)"
```
Comment (mangelajo) [inline config/samples/submariner_v1alpha1_submariner.yaml]: Same about the API server

File: config/rbac/lighthouse-agent/cluster_role_binding.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: submariner-lighthouse-agent
-    namespace: $(SA_NAMESPACE)
+    namespace: placeholder
```
Comment (mangelajo) [inline config/rbac/lighthouse-agent/cluster_role_binding.yaml:9]: why placeholder? what does that mean.

File: config/broker/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+namespace: submariner-k8s-broker
+
+resources:
+  - namespace.yaml        # TODO: cannot define a namespace object in bundle/manifests
+  - ../rbac/broker-admin/service_account.yaml
```
Comment (mangelajo) [inline config/broker/kustomization.yaml]: Ok, this is something that has to happen inside the broker controller:
https://github.com/submariner-io/submariner-operator/blob/devel/controllers/submariner/broker_controller.go#L81

So the user creates the broker namespace, a broker CR, and then the operator will create: Roles, bindings, SAs, etc...

@skitt what do you think?

May be for now it's ok to have one example/bundle. But the user would need to customize the namespace, I guess.

File: config/manifests/bases/submariner.clusterserviceversion.yaml [hunk 7]
```diff
@@ -48,61 +48,67 @@ spec:
...
-    * **debug**: *'true'* to enable logging debugging information;
+    On the broker cluster:
+
+    1. Create the broker namespace `submariner-k8s-broker`.
+    2. Install the submariner operator from the catalog.
+    3. Create a `Broker` CR.
+
+    On connected clusters:
+
+    1. Create the broker namespace `submariner-k8s-broker`.
```
Comment (mangelajo) [inline config/manifests/bases/submariner.clusterserviceversion.yaml]: It shouldn't be necessary to create the broker namespace in connected clusters.

We need to fix our bundles to avoid that. (or keep them separate for now) (see my other comments about the submariner-operator controller).

**Review by mangelajo — COMMENTED** [review]
See my comments, let's setup a meeting tomorrow with @skitt

File: config/samples/submariner_v1alpha1_servicediscovery.yaml (hunk 1 above)
Comment (SteveMattar) [reply config/samples/submariner_v1alpha1_servicediscovery.yaml:15]: @mangelajo all files under `/config` are templates only. The final result after generating the bundle is located in `/bundle/manifests`.

File: config/samples/submariner_v1alpha1_submariner.yaml (hunk 3 above)
Comment (SteveMattar) [reply config/samples/submariner_v1alpha1_submariner.yaml]: Yeah, I was struggling with whether or not to keep this value as an example.

File: config/rbac/lighthouse-agent/cluster_role_binding.yaml (hunk 5 above)
Comment (SteveMattar) [reply config/rbac/lighthouse-agent/cluster_role_binding.yaml:9]: again its a template

File: config/broker/kustomization.yaml (hunk 6 above)
Comment (SteveMattar) [reply config/broker/kustomization.yaml]: Same for the scc definition. I will update the description again.

File: config/manifests/bases/submariner.clusterserviceversion.yaml (hunk 7 above)
Comment (SteveMattar) [reply config/manifests/bases/submariner.clusterserviceversion.yaml]: @mangelajo you are right but the reason we have to do it is because the bundle includes the broker manifests as well and when the users installs the operator on any cluster OLM will automatically deploy all manifests including the broker's. There is no way to add a condition when using this approach. As you mentioned in the previous comment, if the operator handles the namespace and the rest of the broker resources then we don't need this step.

File: config/samples/submariner_v1alpha1_servicediscovery.yaml (hunk 2 above)
Comment (SteveMattar) [reply config/samples/submariner_v1alpha1_servicediscovery.yaml]: My first thought was let's avoid non-global values but I tend to agree.

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 8]
```diff
@@ -94,6 +104,18 @@ spec:
     ### Prerequisites
     Submariner has a few requirements to get started, all are documented [here](https://submariner.io/getting-started/quickstart/kind/#prerequisites).
 
+    For an Openshift cluster perform the following steps to bind the submariner service accounts to the **privileged** SCC.
+
+    ```shell
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-operator
```
Comment (mangelajo) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
```

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 9]
```diff
@@ -94,6 +104,18 @@ spec:
...
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-gateway
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-routeagent
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-globalnet
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-networkplugin-syncer
```
Comment (mangelajo) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
```

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 10]
```diff
@@ -94,6 +104,18 @@ spec:
...
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-lighthouse-agent
```
Comment (mangelajo) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
```

File: config/manifests/bases/submariner.clusterserviceversion.yaml (hunk 7 above)
Comment (SteveMattar) [reply config/manifests/bases/submariner.clusterserviceversion.yaml]: The user should now manually apply the broker manifests.
The description was updated accordingly.

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 11]
```diff
@@ -94,6 +104,15 @@ spec:
     ### Prerequisites
     Submariner has a few requirements to get started, all are documented [here](https://submariner.io/getting-started/quickstart/kind/#prerequisites).
 
+    If running on an OpenShift cluster perform the following steps to bind the submariner service accounts to the **privileged** SCC.
```
Comment (nyechiel) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
    If running on an OpenShift cluster, perform the following steps to bind the Submariner service accounts to the **privileged** SCC.
```

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 12]
```diff
@@ -103,61 +122,68 @@ spec:
...
-    Deploy the broker:
+    On the broker cluster:
+    1. Create the broker namespace called `submariner-k8s-broker`.
+    2. Apply the broker RBAC definitions:
 
-    `subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER>`
+        `kubectl apply -k https://github.com/submariner-io/submariner-operator/tree/release-0.9/config/broker`
 
-    For each cluster you want to join, issue the following command:
+    3. Install the submariner operator from the catalog.
```
Comment (nyechiel) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
    3. Install the Submariner Operator from the catalog.
```

Comment (nyechiel) [reply bundle/manifests/submariner.clusterserviceversion.yaml]: Should we provide a link?

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 13]
```diff
@@ -103,61 +122,68 @@ spec:
...
+    4. Create a new `Broker` custom resource named **submariner-broker**.
 
-    `subctl join --kubeconfig <PATH-TO-JOINING-CLUSTER> broker-info.subm --clusterid <ID>`
+    On connected clusters:
+    1. Install the submariner operator from the catalog.
```
Comment (nyechiel) [inline bundle/manifests/submariner.clusterserviceversion.yaml]:
```suggestion
    1. Install the Submariner Operator from the catalog.
```

Comment (nyechiel) [reply bundle/manifests/submariner.clusterserviceversion.yaml]: Ditto

File: packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml [hunk 14]
```diff
@@ -94,6 +104,15 @@ spec:
     ### Prerequisites
     Submariner has a few requirements to get started, all are documented [here](https://submariner.io/getting-started/quickstart/kind/#prerequisites).
 
+    If running on an OpenShift cluster perform the following steps to bind the submariner service accounts to the **privileged** SCC.
```
Comment (nyechiel) [inline packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml]:
```suggestion
    If running on an OpenShift cluster, perform the following steps to bind the Submariner service accounts to the **privileged** SCC.
```

File: packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml [hunk 15]
```diff
@@ -103,61 +122,68 @@ spec:
...
-    Deploy the broker:
+    On the broker cluster:
+    1. Create the broker namespace called `submariner-k8s-broker`.
+    2. Apply the broker RBAC definitions:
 
-    `subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER>`
+        `kubectl apply -k https://github.com/submariner-io/submariner-operator/tree/release-0.9/config/broker`
 
-    For each cluster you want to join, issue the following command:
+    3. Install the submariner operator from the catalog.
```
Comment (nyechiel) [inline packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml]:
```suggestion
    3. Install the Submariner Operator from the catalog.
```

File: packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml [hunk 16]
```diff
@@ -103,61 +122,68 @@ spec:
...
+    4. Create a new `Broker` custom resource named **submariner-broker**.
 
-    `subctl join --kubeconfig <PATH-TO-JOINING-CLUSTER> broker-info.subm --clusterid <ID>`
+    On connected clusters:
+    1. Install the submariner operator from the catalog.
```
Comment (nyechiel) [inline packagemanifests/0.9.0/submariner.v0.9.0.clusterserviceversion.yaml]:
```suggestion
    1. Install the Submariner Operator from the catalog.
```

File: bundle/manifests/submariner.clusterserviceversion.yaml (hunk 12 above)
Comment (SteveMattar) [reply bundle/manifests/submariner.clusterserviceversion.yaml]: The description appears on the catalog web page. The user is already on the right place and just needs to hit the install button.

**Comment by nyechiel** [comment]
@skitt @mangelajo can you please review?

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 17]
```diff
@@ -94,6 +103,15 @@ spec:
...
 
+    If running on an OpenShift cluster, perform the following steps to bind the Submariner service accounts to the **privileged** SCC.
+
+    ```shell
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-gateway
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-routeagent
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-globalnet
+    oc adm policy add-scc-to-user privileged system:serviceaccount:submariner-operator:submariner-lighthouse-coredns
+    ```
+
```
Comment (mangelajo) [inline bundle/manifests/submariner.clusterserviceversion.yaml:114]: Can we move these under the suggested "#### Deployment with the operator (without subctl)" ?  as those wouldn't apply to subctl?

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 18]
```diff
@@ -103,61 +121,71 @@ spec:
...
 
-    #### Deploying with subctl
-    Install the tool:
-    ```bash
-    curl -Ls https://get.submariner.io | bash
-    export PATH=$PATH:~/.local/bin
-    echo export PATH=\$PATH:~/.local/bin >> ~/.profile
-    ```
-    Deploy the broker:
+    On the broker cluster:
```
Comment (mangelajo) [inline bundle/manifests/submariner.clusterserviceversion.yaml:124]:
```suggestion
#### Deployment with subctl
For deploying the submariner operator and submariner we recommend the use of subctl. Please have a look at our [quickstarts](https://submariner.io/getting-started/quickstart/) for more details.

#### Deployment with the operator (without subctl)
    On the broker cluster:
```

**Review by mangelajo — CHANGES_REQUESTED** [review]
I think I would like to keep a pointer still to subctl, before anybody ventures into standalone operator management they should know that subctl is available and that we prefer that solution instead.

File: bundle/manifests/submariner.clusterserviceversion.yaml [hunk 19]
```diff
@@ -103,61 +121,71 @@ spec:
     For complete information about subctl, please refer to [this page](https://submariner.io/operations/deployment/subctl).
     In addition to Operator and subctl, Submariner also provides [Helm Charts](https://submariner.io/operations/deployment/helm).
```
Comment (SteveMattar) [inline bundle/manifests/submariner.clusterserviceversion.yaml:123]:
```suggestion
### Deployment
    Submariner provides an [Operator](https://github.com/submariner-io/submariner-operator) for easy API-based
    installation and management.
    A command line utility, [subctl](https://github.com/submariner-io/submariner-operator/releases), wraps the
    Operator to aid users with manual deployments and easy experimentation.
    subctl greatly simplifies the deployment of Submariner, and is therefore the recommended deployment method.
    For complete information about subctl, please refer to [this page](https://submariner.io/operations/deployment/subctl).
    In addition to Operator and subctl, Submariner also provides [Helm Charts](https://submariner.io/operations/deployment/helm).
```
@mangelajo `subctl` is mentioned here. There is a reference with a link for all sort of deployments we provide: subctl, helm,...
I think this specific description should focus on the bundle installation only and if the user wants to use a different method he should follow our documentation.
Does it make sense?

Comment (mangelajo) [reply bundle/manifests/submariner.clusterserviceversion.yaml:123]: I want to highlight that they should use subctl as first option, unless they have a good reason to use the operator standalone. As it will save many general issues.

Do you think there is a better way to highlight it?

**Review by skitt — APPROVED** [review]

File: bundle/manifests/submariner.clusterserviceversion.yaml (hunk 19 above)
Comment (SteveMattar) [reply bundle/manifests/submariner.clusterserviceversion.yaml:123]: In our current description we recommend for subctl:

> subctl greatly simplifies the deployment of Submariner, and is therefore the recommended deployment method

We can maybe rephrase the sentence to make the message even more clear.

**Review by dfarrell07 — APPROVED** [review]

File: bundle/manifests/submariner.clusterserviceversion.yaml (hunk 19 above)
Comment (SteveMattar) [reply bundle/manifests/submariner.clusterserviceversion.yaml:123]: @mangelajo

**Review by mangelajo — APPROVED** [review]
